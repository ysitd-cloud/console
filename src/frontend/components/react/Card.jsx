import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export default function Card({
  children,
  color,
  height,
}) {
  const className = classNames(color, 'card');
  const style = {
    // eslint-disable-next-line no-restricted-globals
    height: isNaN(height) ? height : `${height}px`,
  };

  return <div className={className} style={style}>{children}</div>;
}

Card.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

Card.defaultProps = {
  children: null,
  color: '',
  height: 'auto',
};
