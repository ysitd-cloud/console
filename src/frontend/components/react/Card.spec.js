import React from 'react';
import renderer from 'react-test-renderer';
import Card from './Card';

/* globals describe, expect, it */

describe('React Card Component', () => {
  it('match snapshot', () => {
    const component = renderer.create(<Card color="primary">foo</Card>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
