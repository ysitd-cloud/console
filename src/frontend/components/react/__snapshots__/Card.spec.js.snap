// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Card Component match snapshot 1`] = `
<div
  className="primary card"
  style={
    Object {
      "height": "auto",
    }
  }
>
  foo
</div>
`;
